server/master/src/main/java/org/apache/accumulo/master/tableOps/bulkVer2/LoadFiles.java

```diff
@@ -340,27 +341,50 @@ private long loadFiles(TableId tableId, Path bulkDir, LoadMappingIterator loadMa
     return sleepTime;
   }
 
+  private static final Comparator<Text> PREV_COMP = Comparator.nullsFirst(Text::compareTo);
+  private static final Comparator<Text> END_COMP = Comparator.nullsLast(Text::compareTo);
+
   /**
    * Find all the tablets within the provided bulk load mapping range.
    */
   private List<TabletMetadata> findOverlappingTablets(KeyExtent loadRange,
       Iterator<TabletMetadata> tabletIter) {
-    List<TabletMetadata> tablets = new ArrayList<>();
-    TabletMetadata currentTablet = tabletIter.next();
+    try {
 
-    // skip tablets until we find the prevEndRow of loadRange
-    while (!Objects.equals(currentTablet.getPrevEndRow(), loadRange.getPrevEndRow())) {
-      currentTablet = tabletIter.next();
-    }
-    // we have found the first tablet in the range, add it to the list
-    tablets.add(currentTablet);
-
-    // find the remaining tablets within the loadRange by
-    // adding tablets to the list until the endRow matches the loadRange
-    while (!Objects.equals(currentTablet.getEndRow(), loadRange.getEndRow())) {
-      currentTablet = tabletIter.next();
-      tablets.add(currentTablet);
+      List<TabletMetadata> tablets = new ArrayList<>();
+      TabletMetadata currTablet = tabletIter.next();
+
+      int cmp;
+
+      // skip tablets until we find the prevEndRow of loadRange
+      while ((cmp = PREV_COMP.compare(currTablet.getPrevEndRow(), loadRange.getPrevEndRow())) < 0) {
+        currTablet = tabletIter.next();
+      }
+
+      if (cmp != 0) {
+        throw new IllegalStateException("Unexpected prev end row " + currTablet + " " + loadRange);
+      }
+
+      // we have found the first tablet in the range, add it to the list
+      tablets.add(currTablet);
+
+      // find the remaining tablets within the loadRange by
+      // adding tablets to the list until the endRow matches the loadRange
+      while ((cmp = END_COMP.compare(currTablet.getEndRow(), loadRange.getEndRow())) < 0) {
+        currTablet = tabletIter.next();
+        tablets.add(currTablet);
+      }
+
+      if (cmp != 0) {
+        throw new IllegalStateException("Unexpected end row " + currTablet + " " + loadRange);
+      }
+
+      return tablets;
+    } catch (NoSuchElementException e) {
+      NoSuchElementException ne2 =
+          new NoSuchElementException("Failed to find overlapping tablets for " + loadRange);
```

Comment:
Could also include the `currTablet` here, just like in the above `IllegalStateException` (would have to declare the variable outside the try, of course). Also, instead of throwing `NoSuchElementException` here, with `.initCause`, could `throw new IllegalStateException(msg, e);`